tasks/kmt.py

```diff
@@ -269,55 +272,141 @@ def full_arch(arch):
     return arch
 
 
+def build_tests_package(ctx, source_dir, stack, arch, ci, verbose=True):
+    root = os.path.join(source_dir, "kmt-deps")
+    test_archive = f"tests-{arch}.tar.gz"
+    if not ci:
+        system_probe_tests = os.path.join(root, stack, "opt/system-probe-tests")
+        test_pkgs = os.path.join(
+            source_dir, "test/kitchen/site-cookbooks/dd-system-probe-check/files/default/tests/pkg"
+        )
+        ctx.run(f"rm -rf {system_probe_tests} && mkdir -p {system_probe_tests}", hide=(not verbose))
+        ctx.run(f"cp -R {test_pkgs} {system_probe_tests}", hide=(not verbose))
+        with ctx.cd(os.path.join(root, stack)):
+            ctx.run(f"tar czvf {test_archive} opt", hide=(not verbose))
+
+
 @task
-def prepare(ctx, vms, stack=None, arch=None, ssh_key=None, rebuild_deps=False, packages="", verbose=True):
+def build_dependencies(ctx, arch, layout_file, source_dir, ci=False, stack=None, verbose=True):
+    root = os.path.join(source_dir, "kmt-deps")
+    deps_dir = os.path.join(root, "dependencies")
+    if not ci:
+        if stack is None:
+            raise Exit("no stack name provided")
+        deps_dir = os.path.join(root, stack, "dependencies")
+        # in the CI we can rely on gotestsum being present
+        download_gotestsum(ctx)
+
+    if os.path.exists(deps_dir):
```

Comment:
which `deps_dir` do you want to clean-up, the `root/dependencies` or the `root/stack/dependencies`? Just double checking

Reply:
Either. The layout will be slightly different locally vs the CI. The reasons is that you can have multiple "stacks" running locally, whereas there is no concept of a "stack" in the CI.
Depending on whether `ci` is set we want to remove the correct one.